fl4health/preprocessing/autoencoders/dim_reduction.py

```diff
@@ -26,59 +28,113 @@ def __repr__(self) -> str:
+        Transformer processor to encode the data using VAE encoder.
+
+        Args:
+            checkpointing_path (Path): Path to the saved model.
+            return_mu (bool, optional): If true, only mu is returned. Defaults to False.
+        """
         super().__init__(checkpointing_path)
-        self.include_variance = include_variance
+        self.return_mu = return_mu
 
     def __call__(self, sample: torch.Tensor) -> torch.Tensor:
         # This transformer is called for the input samples after they are transfered into torch tensors.
         mu, logvar = self.autoencoder.encode(sample.to(self.device))
-        if self.include_variance:
-            return torch.cat(mu.clone().detach(), logvar.clone().detach())
-        return mu.clone().detach()
+        if self.return_mu:
+            return mu.clone().detach()
+        # By default returns cat(mu,logvar).
+        # Contatation is performed on the last dimention which for both the batched data and single data
+        # is the latent space dimention.
+        return torch.cat((mu.clone().detach(), logvar.clone().detach()), dim=-1)
 
 
-class CvaeFixedConditionProcessor(Processing):
+class CvaeFixedConditionProcessor(AutoEncoderProcessing):
     """Transformer processor to encode the data using CVAE encoder with client-specific condition."""
 
     def __init__(
         self,
         checkpointing_path: Path,
         condition: torch.Tensor,
-        include_variance: bool = False,
+        return_mu: bool = False,
+        batch_size: int = 1,
     ) -> None:
+        super().__init__(checkpointing_path)
         self.condition = condition
-        self.include_variance = include_variance
+        self.return_mu = return_mu
         assert (
             self.condition.dim() == 1
         ), f"Error: condition should be a 1D vector instead of a {self.condition.dim()}D tensor."
-        super().__init__(checkpointing_path)
+        # Assuming data is "batch first"
+        # If we are processing a batch of data, condition should be repeated for each sample.
+        if batch_size > 1:
+            self.condition = self.condition.repeat(batch_size, 1)
 
     def __call__(self, sample: torch.Tensor) -> torch.Tensor:
         # This transformer is called for the input samples after they are transfered into torch tensors.
         mu, logvar = self.autoencoder.encode(sample.to(self.device), self.condition.to(self.device))
-        if self.include_variance:
-            return torch.cat(mu.clone().detach(), logvar.clone().detach())
-        return mu.clone().detach()
+        if self.return_mu:
+            return mu.clone().detach()
+        # By default returns cat(mu,logvar)
+        # Contatation is performed on the last dimention which for both the batched data and single data
+        # is the latent space dimention.
+        return torch.cat((mu.clone().detach(), logvar.clone().detach()), dim=-1)
+
+
+class CvaeNonFixedConditionProcessor(AutoEncoderProcessing):
```

Comment:
Super minor, but rather than "NonFixedCondition" perhaps we can use the word VariableCondition?